libvast/builtins/operators/top_rare.cpp

```diff
@@ -0,0 +1,86 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/argument_parser.hpp>
+#include <vast/concept/parseable/string/char_class.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
+#include <vast/detail/string_literal.hpp>
+#include <vast/error.hpp>
+#include <vast/location.hpp>
+#include <vast/logger.hpp>
+#include <vast/pipeline.hpp>
+#include <vast/plugin.hpp>
+
+namespace vast::plugins::top_rare {
+
+namespace {
+
+template <detail::string_literal Name, detail::string_literal SortOrder>
+class top_rare_plugin final : public virtual operator_parser_plugin {
+  auto name() const -> std::string override {
+    return std::string{Name.str()};
+  }
+
+  auto parse_operator(parser_interface& p) const -> operator_ptr override {
+    auto parser = argument_parser{
+      std::string{Name.str()}, fmt::format("https://docs.tenzir.com/docs/next/"
+                                           "operators/transformations/{}",
+                                           Name.str())};
+    auto field = located<std::string>{};
+    auto count_field = std::optional<located<std::string>>{};
+    parser.add(field, "<str>");
+    parser.add("-c,--count-field", count_field, "<str>");
+    parser.parse(p);
+    if (count_field) {
+      if (count_field->inner.empty()) {
+        diagnostic::error("need a string value for `count-field` parameter")
+          .primary(field.source)
```

Comment:
One suggestion, and one fix :)

```suggestion
        diagnostic::error("`--count-field` must not be empty")
          .primary(count_field->source)
```